lnwallet/channel.go

```diff
@@ -3284,69 +3306,208 @@ func (lc *LightningChannel) ChanSyncMsg() (*lnwire.ChannelReestablish, error) {
+	ourBalance := commitChain.tail().ourBalance
+	theirBalance := commitChain.tail().theirBalance
 
-	// Run through all the HTLCs that will be covered by this transaction
-	// in order to calculate theirs count.
-	view := lc.fetchHTLCView(theirLogCounter, ourLogCounter)
+	// Add the fee from the previous commitment state back to the
+	// initiator's balance, so that the fee can be recalculated and
+	// re-applied in case fee estimation parameters have changed or
+	// the number of outstanding HTLCs has changed.
+	if lc.channelState.IsInitiator {
+		ourBalance += lnwire.NewMSatFromSatoshis(
+			commitChain.tail().fee)
+	} else if !lc.channelState.IsInitiator {
+		theirBalance += lnwire.NewMSatFromSatoshis(
+			commitChain.tail().fee)
+	}
+	nextHeight := commitChain.tail().height + 1
 
-	if remote {
-		for _, entry := range view.theirUpdates {
-			if entry.EntryType == Add {
-				htlcCount++
-			}
+	// We evaluate the view at this stage, meaning settled and
+	// failed HTLCs will remove their corresponding added HTLCs.
+	// The resulting filtered view will only have Add entries left,
+	// making it easy to compare the channel constraints to the
+	// final commitment state.
+	ourInitialBalance := ourBalance
+	theirInitialBalance := theirBalance
+	filteredHTLCView := lc.evaluateHTLCView(view, &ourBalance,
+		&theirBalance, nextHeight, remoteChain, false)
+
+	// TODO(halseth): the following code segment is duplicated in
+	// fetchCommitmentView, change that.
+
+	// Initiate feePerKw to the last committed fee for this chain as we'll
+	// need this to determine which HTLC's are dust, and also the final fee
+	// rate.
+	feePerKw := commitChain.tail().feePerKw
+
+	// Check if any fee updates have taken place since that last
+	// commitment.
+	if lc.channelState.IsInitiator {
+		switch {
+		// We've sent an update_fee message since our last commitment,
+		// and now are now creating a commitment that reflects the new
+		// fee update.
+		case remoteChain && lc.pendingFeeUpdate != nil:
+			feePerKw = *lc.pendingFeeUpdate
+
+		// We've created a new commitment for the remote chain that
+		// includes a fee update, and have not received a commitment
+		// after the fee update has been ACKed.
+		case !remoteChain && lc.pendingAckFeeUpdate != nil:
+			feePerKw = *lc.pendingAckFeeUpdate
 		}
-		for _, entry := range view.ourUpdates {
-			if entry.EntryType != Add {
-				htlcCount--
-			}
+	} else {
+		switch {
+		// We've received a fee update since the last local commitment,
+		// so we'll include the fee update in the current view.
+		case !remoteChain && lc.pendingFeeUpdate != nil:
+			feePerKw = *lc.pendingFeeUpdate
+
+		// Earlier we received a commitment that signed an earlier fee
+		// update, and now we must ACK that update.
+		case remoteChain && lc.pendingAckFeeUpdate != nil:
+			feePerKw = *lc.pendingAckFeeUpdate
 		}
 	}
 
-	if local {
-		for _, entry := range view.ourUpdates {
+	// Now go through all HTLCs at this stage, to calculate the total
```

Comment:
Seems like instead, the `availableBalance` could could've bee modified slightly to avoid this additional duplication?